src/firefly/js/visualize/ui/CatalogSelectViewPanel.jsx

```diff
@@ -0,0 +1,623 @@
+import {VoSearchPanel} from '../../ui/VoSearchPanel.jsx';
+import {FileUpload} from '../../ui/FileUpload.jsx';
+import {convertAngle} from '../VisUtil';
+
+import './CatalogTableListField.css';
+
+/**
+ * group key for fieldgroup comp
+ */
+const gkey = 'CATALOG_PANEL';
+
+/**define the helpButton*/
+const helpIdStyle = {'textAlign': 'center', display: 'inline-block', height: 40, marginRight: 20};
+
+const dropdownName = 'IrsaCatalogDropDown';
+
+const initRadiusArcSec = parseFloat(500 / 3600).toString();
+
+/**
+ * Globally scoped here, master table, columns object
+ * @type {Array}
+ */
+var catmaster = [], cols = [];
+
+export class CatalogSelectViewPanel extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {fields: FieldGroupUtils.getGroupFields(gkey)};
+    }
+
+    componentWillUnmount() {
+        this.iAmMounted = false;
+        if (this.unbinder) this.unbinder();
+    }
+
+    componentDidMount() {
+        this.iAmMounted = true;
+        this.unbinder = FieldGroupUtils.bindToStore(gkey, (fields) => {
+            if (fields !== this.state.fields && this.iAmMounted) {
+                this.setState({fields});
+            }
+        });
+    }
+
+    render() {
+        var {fields}= this.state;
+        return (
+            <div style={{padding: 10}}>
+                <FormPanel
+                    groupKey={gkey}
+                    onSubmit={(request) => onSearchSubmit(request)}
+                    onCancel={hideSearchPanel}>
+                    <CatalogSelectView fields={fields}/>
+                </FormPanel>
+            </div>
+        );
+    }
+}
+
+function onSearchSubmit(request) {
+    console.log('request ' + JSON.stringify(request));
+
+    if (request.Tabs === 'catalog') {
+        const wp = parseWorldPt(request[ServerParams.USER_TARGET_WORLD_PT]);
+        if (!wp && (request.spatial === SpatialMethod.Cone.value
+            || request.spatial === SpatialMethod.Box.value
+            || request.spatial === SpatialMethod.Elliptical.value)) {
+            showInfoPopup('Target is required');
+            return;
+        }
+        doCatalog(request);
+    }
+    else if (request.Tabs === 'loadcat') {
+        doLoadTable(request);
+    }
+    else if (request.Tabs === 'vosearch') {
+        doVoSearch(request);
+    }
+    else {
+        console.log('request no supported');
+    }
+}
+function hideSearchPanel() {
+    dispatchHideDropDown();
+}
+
+function doCatalog(request) {
+
+    const conesize = convertAngle('deg', 'arcsec', request.conesize);
+    var title = request.project + '-' + request.cattable + ' (' + request.spatial;
+    if (request.spatial === SpatialMethod.Box.value || request.spatial === SpatialMethod.Cone.value || request.spatial === SpatialMethod.Elliptical.value) {
+        title += ':' + conesize;
+    }
+    title += ')';
+    var tReq = TableRequest.newInstance({
+
+        id: 'GatorQuery',
+        title: title,
```

Comment:
yes, ESLint has a rule for that and warned me but forgot to replace. 
Good catch!

On 5/14/16 11:04 PM, Trey Roby wrote:
>
> In src/firefly/js/visualize/ui/CatalogSelectViewPanel.jsx 
> <https://github.com/Caltech-IPAC/firefly/pull/85#discussion_r63286709>:
>
> > +function hideSearchPanel() {
> > +    dispatchHideDropDown();
> > +}
> > +
> > +function doCatalog(request) {
> > +
> > +    const conesize = convertAngle('deg', 'arcsec', request.conesize);
> > +    var title = request.project + '-' + request.cattable + ' (' + request.spatial;
> > +    if (request.spatial === SpatialMethod.Box.value || request.spatial === SpatialMethod.Cone.value || request.spatial === SpatialMethod.Elliptical.value) {
> > +        title += ':' + conesize;
> > +    }
> > +    title += ')';
> > +    var tReq = TableRequest.newInstance({
> > +
> > +        id: 'GatorQuery',
> > +        title: title,
>
> could be just |title|
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/Caltech-IPAC/firefly/pull/85/files/8bdfd737fd8631b45d10aff1ebb2742bee675e9e#r63286709>
>

-- 
Emmanuel Joliet
NASA/IPAC  Infrared Science Archive
MS 100-22, Caltech, Pasadena, CA 91125
Office: 170
Email:  ejoliet@ipac.caltech.edu
Phone:  (626) 395-1489